designcompose/src/main/java/com/android/designcompose/Utils.kt

```diff
@@ -1276,41 +1278,53 @@ internal constructor(
  *
  * XXX: Doesn't consider transform origin.
  */
-internal fun Optional<List<Float>>.asComposeTransform(
+internal fun Optional<LayoutTransform>.asComposeTransform(
     density: Float
 ): androidx.compose.ui.graphics.Matrix? {
     return map {
-            if (it.size != 16) {
+            val transform =
+                androidx.compose.ui.graphics.Matrix(
+                    floatArrayOf(
+                        it.m11,
+                        it.m12,
+                        it.m13,
+                        it.m14,
+                        it.m21,
+                        it.m22,
+                        it.m23,
+                        it.m24,
+                        it.m31,
+                        it.m32,
+                        it.m33,
+                        it.m34,
+                        it.m41,
+                        it.m42,
+                        it.m43,
+                        it.m44
+                    )
+                )
+            if (transform.isIdentity()) {
                 null
             } else {
-                val transform = androidx.compose.ui.graphics.Matrix(it.toFloatArray())
-                if (transform.isIdentity()) {
-                    null
-                } else {
-                    val adjust = androidx.compose.ui.graphics.Matrix()
-                    adjust.scale(1.0f / density, 1.0f / density, 1.0f / density)
-                    adjust.timesAssign(transform)
-                    val unadjust = androidx.compose.ui.graphics.Matrix()
-                    unadjust.scale(density, density, density)
-                    adjust.timesAssign(unadjust)
-                    adjust
-                }
+                val adjust = androidx.compose.ui.graphics.Matrix()
+                adjust.scale(1.0f / density, 1.0f / density, 1.0f / density)
+                adjust.timesAssign(transform)
+                val unadjust = androidx.compose.ui.graphics.Matrix()
+                unadjust.scale(density, density, density)
+                adjust.timesAssign(unadjust)
+                adjust
             }
         }
         .orElse(null)
 }
 
-internal fun Optional<List<Float>>.asSkiaMatrix(): Matrix? {
+internal fun Optional<AffineTransform>.asSkiaMatrix(): Matrix? {
     return map {
-            if (it.size != 6) {
-                null
-            } else {
-                val skMatrix = Matrix()
-                skMatrix.setValues(
-                    floatArrayOf(it[0], it[1], it[4], it[2], it[3], it[5], 0.0f, 0.0f, 1.0f)
-                )
-                skMatrix
-            }
+            val skMatrix = Matrix()
+            skMatrix.setValues(
+                floatArrayOf(it.m11, it.m12, it.m31, it.m21, it.m22, it.m32, 0.0f, 0.0f, 1.0f)
```

Comment:
How did you do this conversion? I would have thought it woudl be `m11, m12, m21, m13, m14, m22, 0.0f, 0.0f, 1.0f`

Reply:
AffineTransform has 
m11, m12,
m21, m22,
m31, m32
and it used to be it[0], it[1], it[4], it[2], it[3], it[5]

I don't know what it really does though...